src/java/org/apache/cassandra/streaming/StreamSession.java

```diff
@@ -981,7 +988,7 @@ public void streamSent(OutgoingStreamMessage message)
         StreamTransferTask task = transfers.get(message.header.tableId);
         if (task != null)
         {
-            task.scheduleTimeout(message.header.sequenceNumber, 12, TimeUnit.HOURS);
+            task.scheduleTimeout(message.header.sequenceNumber, DatabaseDescriptor.timeoutDelay(), TimeUnit.MILLISECONDS);
```

Comment:
this is brittle, rather than expecting the config to be MILLISECONDS, we should have the DD return the Duration and convert to nanos; this allows users to define any value they want.